src/main/java/net/coderbot/iris/pipeline/TransformPatcher.java

```diff
@@ -0,0 +1,512 @@
+
+					@Override
+					public void enterMultiplicativeExpression(MultiplicativeExpressionContext ctx) {
+						checkPatternMatch(bareMultPattern, ctx, (match) -> replaceNode(
+								ctx, "vec4(" + lightmapCoordsExpression + ", 0.0, 1.0)", GLSLParser::expression));
+
+						// PREV NOTE
+						// NB: Technically this isn't a correct transformation (it changes the values
+						// slightly), however the shader code being replaced isn't correct to begin with
+						// since it doesn't properly apply the centering / scaling transformation like
+						// gl_TextureMatrix[1] would. Therefore, I think this is acceptable.
+						// This code shows up in Sildur's shaderpacks.
+						ParseTreeMatch match = extraPattern.match(ctx);
+						if (match.succeeded()) {
+							String texCoord = match.get("texCoord").getText();
+							if (texCoord.equals("gl_MultiTexCoord1") || texCoord.equals("gl_MultiTexCoord2")) {
+								replaceNode(ctx, lightmapCoordsExpression, GLSLParser::expression);
+							}
+						}
+					}
+
+					@Override
+					public void enterArrayAccessExpression(ArrayAccessExpressionContext ctx) {
+						ParseTreeMatch match = textureMatrixPattern.match(ctx);
+						if (match.succeeded()) {
+							String texMatrixIndex = match.get("texMatrixIndex").getText();
+							if (texMatrixIndex.equals("1") || texMatrixIndex.equals("2")) {
+								replaceNode(ctx, new StringNode(irisLightmapTexMat));
+								needsLightmapTexMatInjection = true;
+							}
+						}
+					}
+				});
+
+				chainDependent(new RunPhase<Parameters>() {
+					@Override
+					protected void run(TranslationUnitContext ctx) {
+						injectExternalDeclaration(InjectionPoint.BEFORE_FUNCTIONS, "uniform mat4 iris_LightmapTextureMatrix;");
+					}
+
+					@Override
+					public boolean isActive() {
+						return needsLightmapTexMatInjection;
+					}
+				});
+
+				chainConcurrentDependent(
+						RunPhase.withInjectExternalDeclarations(InjectionPoint.BEFORE_FUNCTIONS, "attribute vec2 a_LightCoord;"));
+
+				chainConcurrentDependent(new SearchTerminals<Parameters>()
+						.addTarget(new ParsedReplaceTargetImpl<>("gl_MultiTexCoord1",
+								texCoordFallbackReplacement, GLSLParser::expression))
+						.addTarget(new ParsedReplaceTargetImpl<>("gl_MultiTexCoord2",
+								texCoordFallbackReplacement, GLSLParser::expression)));
+			}
+		};
+		// #endregion patchSodiumTerrain
+
+		manager = new TransformationManager<Parameters>(new Transformation<Parameters>() {
+			@Override
+			protected void setupGraph() {
+				Patch patch = getJobParameters().patch;
+				ShaderType type = getJobParameters().type;
+
+				addEndDependent(detectReserved);
+
+				// patchAttributes
+				if (patch == Patch.ATTRIBUTES) {
+					addEndDependent(replaceEntityColorDeclaration);
+
+					if (type == ShaderType.VERTEX || type == ShaderType.GEOMETRY) {
+						addEndDependent(wrapOverlay);
+					} else if (type == ShaderType.FRAGMENT) {
+						addEndDependent(renameEntityColorFragment);
+					}
+				}
```

Comment:
This seems to be missing a lot of the new additions to AttributeShaderTransformer that happened in the bulletproof-state-tracking branch

Reply:
I have not yet updated to the newest version of `AttributeShaderTransformer`.